hepcrawl/spiders/aps_spider.py

```diff
@@ -76,53 +77,24 @@ def start_requests(self):
         yield Request(self.url)
 
     def parse(self, response):
-        """Parse a APS JSON file into a HEP record."""
+        """Parse a APS record into a HEP record.
+
+        Attempts to parse an XML JATS full text first, if available, and falls
+        back to parsing JSON if such is not available.
+        """
         aps_response = json.loads(response.body_as_unicode())
 
         for article in aps_response['data']:
-            record = HEPLoader(item=HEPRecord(), response=response)
-
-            dois = [get_nested(article, 'identifiers', 'doi')]
-            record.add_dois(dois_values=dois)
-            record.add_value('page_nr', str(article.get('numPages', '')))
-
-            record.add_value('abstract', get_nested(article, 'abstract', 'value'))
-            record.add_value('title', get_nested(article, 'title', 'value'))
-            # record.add_value('subtitle', '')
-
-            authors, collaborations = self._get_authors_and_collab(article)
-            record.add_value('authors', authors)
-            record.add_value('collaborations', collaborations)
-
-            # record.add_value('free_keywords', free_keywords)
-            # record.add_value('classification_numbers', classification_numbers)
-
-            record.add_value('journal_title', get_nested(article, 'journal', 'abbreviatedName'))
-            record.add_value('journal_issue', get_nested(article, 'issue', 'number'))
-            record.add_value('journal_volume', get_nested(article, 'volume', 'number'))
-            # record.add_value('journal_artid', )
+            doi = get_nested(article, 'identifiers', 'doi')
 
-            published_date = article.get('date', '')
-            record.add_value('journal_year', int(published_date[:4]))
-            record.add_value('date_published', published_date)
-            record.add_value('copyright_holder', get_nested(article, 'rights', 'copyrightHolders')[0]['name'])
-            record.add_value('copyright_year', str(get_nested(article, 'rights', 'copyrightYear')))
-            record.add_value('copyright_statement', get_nested(article, 'rights', 'rightsStatement'))
-            record.add_value('copyright_material', 'Article')
-
-            license = get_licenses(
-                license_url=get_nested(article, 'rights', 'licenses')[0]['url']
-            )
-            record.add_value('license', license)
-
-            record.add_value('collections', ['HEP', 'Citeable', 'Published'])
-
-            parsed_item = ParsedItem(
-                record=record.load_item(),
-                record_format='hepcrawl',
-            )
-
-            yield parsed_item
+            if doi:
+                request = Request(url='{}/{}'.format(self.aps_base_url, doi),
```

Comment:
you should also attach the `xml` file (as a `hidden` `fulltext`) to the record with `add_document`.

Reply:
How do I know the `key`? Do I have to attach the file myself using `invenio-records-files`, or do we already have a facility for this somewhere in Inspire?

Reply:
the key should be the file name